.github/actions/bootstrap/action.yaml

```diff
@@ -12,49 +12,66 @@ inputs:
 outputs:
   cache-hit:
     description: "Bootstrap environment was restored from cache"
-    value: ${{ fromJSON(steps.bootstrap-cache.outputs.outputs).cache-hit }} # retry returns all outputs in `outputs`
+    value: ${{ fromJSON(steps.restore.outputs.outputs).cache-hit }} # dynamic action returns all outputs in `outputs`
 
 runs:
   using: "composite"
   steps:
-    - name: Calculate bootstrap cache key
-      id: bootstrap-cache-key
+    - name: Determine bootstrap cache configuration
+      id: prepare
       shell: bash
       run: |
-        # Calculate bootstrap cache key
+        # Determine bootstrap cache configuration
         # In addition to the various setup files, the work directory matters as well,
         # because the bootstrapped Pigweed environment contains absolute paths.
+        echo "Calculating bootstrap cache key for '$PWD'"
         FILES_HASH="${{ hashFiles('scripts/setup/*', 'third_party/pigweed/**') }}"
         FINAL_HASH="$(echo "$PWD:$FILES_HASH" | shasum -a 256 | cut -d' ' -f1)"
-        echo "Calculated bootstrap cache key for '$PWD': $FINAL_HASH"
-        echo "hash=$FINAL_HASH" >> "$GITHUB_OUTPUT"
+        echo key="${RUNNER_OS}-${RUNNER_ARCH}-${{ inputs.platform }}-${FINAL_HASH}" | tee -a "$GITHUB_OUTPUT"
 
-    - uses: Wandalen/wretry.action@v1.4.10
-      name: Bootstrap from cache
-      id: bootstrap-cache
+        # Split caches across backends
```

Comment:
This could use comments explaining _why_ we are doing that.   Because I don't understand why we're doing that.